core/modules/replica/SqlRemoveTablePartitionsJob.cc

```diff
@@ -0,0 +1,194 @@
+    list<pair<string,string>> result;
+    result.emplace_back("database", database());
+    result.emplace_back("table", table());
+    result.emplace_back("all_workers", string(allWorkers() ? "1" : "0"));
+    return result;
+}
+
+
+list<SqlRequest::Ptr> SqlRemoveTablePartitionsJob::launchRequests(util::Lock const& lock,
+                                                                  string const& worker,
+                                                                  size_t maxRequests) {
+    list<SqlRequest::Ptr> requests;
+
+    // Initialize worker's sub-collection if the first time seeing
+    // this worker.
+    if (not _workers2tables.count(worker)) {
+
+        // This table must exist in both versions
+        _workers2tables[worker].push_back(table());
+
+        // Add chunk specific tables
+        if (_isPartitioned) {
+
+            // Locate all chunks registered on the worker. These chunks will be used
+            // to build names of the corresponding chunk-specific partitioned tables.
+
+            vector<ReplicaInfo> replicas;
+            controller()->serviceProvider()->databaseServices()->findWorkerReplicas(
+                replicas,
+                worker,
+                database()
+            );
+            for (auto&& replica: replicas) {
+                auto const chunk = replica.chunk();
+                _workers2tables[worker].push_back(table() + "_" + to_string(chunk));
+                _workers2tables[worker].push_back(table() + "FullOverlap_" + to_string(chunk));
+            }
+        }
+    }
+
+    // Launch up to (not to exceed) the specified number of requests for tables
+    // by pulling table names from the worker's sub-collection. NOte that used
+    // tables will get removed from the sub-collections.
+
+    auto const self = shared_from_base<SqlRemoveTablePartitionsJob>();
+    while (not _workers2tables[worker].empty() and requests.size() < maxRequests) {
+        requests.push_back(
+            controller()->sqlRemoveTablePartitions(
+                worker,
+                database(),
+                _workers2tables[worker].front(),
+                [self] (SqlRemoveTablePartitionsRequest::Ptr const& request) {
+                    self->onRequestFinish(request);
+                },
+                options(lock).priority,
+                true,   /* keepTracking*/
+                id()    /* jobId */
+            )
+        );
+        _workers2tables[worker].pop_front();
+    }
+    return requests;
+}
+
+
+void SqlRemoveTablePartitionsJob::stopRequest(util::Lock const& lock,
+                                              SqlRequest::Ptr const& request) {
+    controller()->stopById<StopSqlRemoveTablePartitionsRequest>(
+        request->worker(),
+        request->id(),
+        nullptr,    /* onFinish */
+        options(lock).priority,
+        true,       /* keepTracking */
+        id()        /* jobId */
+    );
+}
```

Comment:
It looks like stopRequest always takes the same arguments.  If Job takes a template argument, you could get rid of all of them.
```
template <class T>
class Job {
...
    virtual stopRequest(util::Lock const& lock, SqlRequest::Ptr const& request) {
        controller()->stopRequest<T>(
...
}

class SqlRemoveTablePartitionsJob : public Job <StopSqlRemoveTablePartitionsRequest> 
```
Not the best example here as I skipped SqlJob.

Reply:
This is a good suggestion. And it doesn't require introducing a class template. The problem could also be solved by pushing this operation into the existing base class `SqlJob` as a function template:
```
class SqlJob {
    template <class REQUEST>
    void stopRequest(util::Lock const& lock, SqlRequest::Ptr const& request) {
        controller()->stopRequest<REQUEST>(...);
    }

Reply:
refactored